sets/util.go

```diff
@@ -0,0 +1,144 @@
+package sets
+
+// Difference return s1 - s2
+func Difference(s1 Set, s2 Set) Set {
```

Comment:
:microscope:  these should all be methods on `Set` which accept another set as a parameter
Intersection, Union and Equals I could see being functions since they are commutative but since Difference and Contains aren't, I'm of the opinion they should be methods on a set e.g. `mySet.Contains(otherSet)`

Then why not just make em all methods

Reply:
I thought about this. The main reason I avoided it was that it gives you the ability to operate on two sets that might be nil, thus saving the caller from otherwise needing to add a nil check. I definitely see the advantage of putting `Contains` and `Difference` on the `Set` interface though. I'll give it some thought and wait for other opinions.